landlab/graph/framed_voronoi/framed_voronoi.py

```diff
@@ -0,0 +1,483 @@
+        xy_min_spacing: float or tuple of float, optional.
+            Final minimal spacing between nodes. Random moves of the core nodes
+            around their position cannot be above this threshold:
+            (xy_spacing - xy_min_spacing) /2
+            If float, same minimal spacing for x and y.
+        random_seed: bool, optional.
+            If True, the moves of coordinates are completely random. False is used
+            when reproducibility of moves is needed. Move pseudo-randomness is
+            then controlled by the parameter seed.
+        seed: int or tuple of int, optional.
+            Seeds used to generate the random x and y moves. This parameter is unused
+            when random_seed = True.
+
+        Returns
+        -------
+        tuple of ndarray(int).
+            The arrays of x and y coordinates.
+
+        Examples
+        --------
+        >>> from landlab.graph.framed_voronoi.framed_voronoi import HorizontalRectVoronoiGraph
+        >>> HorizontalRectVoronoiGraph.xy_of_node((3, 3), random_seed=False, seed=(200, 500))[0][3:6]       # doctest: +NORMALIZE_WHITESPACE
+        array([ 0.        ,  1.07341707,  2.        ])
+        >>> HorizontalRectVoronoiGraph.xy_of_node((3, 3), random_seed=False, seed=(200, 500))[1][3:6]       # doctest: +NORMALIZE_WHITESPACE
+        array([ 0.749     ,  1.03337157,  1.251     ])
+        """
+
+        n_rows, n_cols = shape
+        max_move = [
+            (xy_spacing[0] - xy_min_spacing[0]) / 2,
+            (xy_spacing[1] - xy_min_spacing[1]) / 2,
+        ]
+        n_nodes = HorizontalRectVoronoiGraph.number_of_nodes(shape)
+        n_core_nodes = n_nodes - HorizontalRectVoronoiGraph.number_of_perimeter_nodes(
+            shape
+        )
+
+        # Generation of a rectangular grid, coordinates must be float
+        x_of_node, y_of_node = np.meshgrid(
+            np.arange(n_cols, dtype=float) * xy_spacing[0] + xy_of_lower_left[0],
+            np.arange(n_rows, dtype=float) * xy_spacing[1] + xy_of_lower_left[1],
+        )
+        # Randomly move the coordinates of the core nodes of the grid. Move below +/- (spacing - min_spacing)/2
+        if random_seed:
+            xy_random_generator = (np.random.default_rng(), np.random.default_rng())
+        else:
+            xy_random_generator = (
+                np.random.Generator(np.random.PCG64(seed[0])),
+                np.random.Generator(np.random.PCG64(seed[1])),
+            )
+
+        x_y_random_move = (
+            xy_random_generator[0].random(n_core_nodes) * max_move[0] * 2 - max_move[0],
+            xy_random_generator[1].random(n_core_nodes) * max_move[1] * 2 - max_move[1],
+        )
+
+        k = 0
+        for i in range(1, n_rows - 1):
+            for j in range(1, n_cols - 1):
+                x_of_node[i, j] += x_y_random_move[0][k]
+                y_of_node[i, j] += x_y_random_move[1][k]
+                k += 1
```

Comment:
I think you can remove these loops with something like,
```python
x_of_node[1:-1, 1:-1] += xy_random_move[0]
y_of_node[1:-1, 1:-1] += xy_random_move[1]
```

Reply:
Done. Actually you have to reshape x_y_random_move so that works.